backend/controller/ingress/response.go

```diff
@@ -63,65 +63,107 @@ func ResponseForVerb(sch *schema.Schema, verb *schema.Verb, response HTTPRespons
 		}
 	}
 
-	switch bodyType := fieldType.(type) {
+	outBody, err := bodyForType(fieldType, sch, body)
+	return outBody, headers, err
+}
+
+func bodyForType(typ schema.Type, sch *schema.Schema, data []byte) ([]byte, error) {
+	switch t := typ.(type) {
 	case *schema.DataRef:
 		var responseMap map[string]any
-		err := json.Unmarshal(body, &responseMap)
+		err := json.Unmarshal(data, &responseMap)
 		if err != nil {
-			return nil, nil, fmt.Errorf("HTTP response body is not valid JSON: %w", err)
+			return nil, fmt.Errorf("HTTP response body is not valid JSON: %w", err)
 		}
 
-		aliasedResponseMap, err := transformToAliasedFields(bodyType, sch, responseMap)
+		aliasedResponseMap, err := transformToAliasedFields(t, sch, responseMap)
 		if err != nil {
-			return nil, nil, err
+			return nil, err
 		}
 		outBody, err := json.Marshal(aliasedResponseMap)
-		return outBody, headers, err
+		return outBody, err
+
+	case *schema.Array:
+		var responseArray []json.RawMessage
+		err := json.Unmarshal(data, &responseArray)
+		if err != nil {
+			return nil, fmt.Errorf("HTTP response body is not valid JSON array: %w", err)
+		}
+
+		transformedArrayData := make([]any, len(responseArray))
+		for i, rawElement := range responseArray {
+			var transformedElement any
+			err := json.Unmarshal(rawElement, &transformedElement)
+			if err != nil {
+				return nil, fmt.Errorf("failed to unmarshal array element: %w", err)
+			}
+
+			// Transform the array element to its aliased form.
+			if dataRef, ok := t.Element.(*schema.DataRef); ok {
```

Comment:
Could this not call bodyForType recursively? What if you have an array of strings? Or an array of arrays?